src/lib.rs

```diff
@@ -15,6 +15,17 @@
 //! }
 //! ```
 //!
+//! However, **make sure to only use the above form with a mutable reference to an existing iterator**,
+//! not with an expression that creates an iterator.
+//! For example, the following code will loop forever over the first element of the array:
+//!
+//! ```ignore
```

Comment:
Nit: let's use `no_run` here rather than `ignore` so the code will still at least be compiled.